codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java

```diff
@@ -603,7 +603,11 @@ public T clear() {
     @Override
     public Boolean getBoolean(K name) {
         V v = get(name);
-        return v != null ? valueConverter.convertToBoolean(v) : null;
+        try {
+            return v != null ? valueConverter.convertToBoolean(v) : null;
```

Comment:
There is another bug here that I haven't filed: `convertToBoolean` has different behavior for `AsciiString` header values than for other `CharSequence` values.

`AsciiString` uses [this null-byte-check conversion](https://github.com/netty/netty/blob/4f982be91e6d61c7c7a9d8d63115e0ebe65b2774/common/src/main/java/io/netty/util/AsciiString.java#L1217-L1219), whereas other strings use [`Boolean.parseBoolean`](https://github.com/netty/netty/blob/4f982be91e6d61c7c7a9d8d63115e0ebe65b2774/codec/src/main/java/io/netty/handler/codec/CharSequenceValueConverter.java#L68-L73), which treats [only strings matching "true"](https://docs.oracle.com/javase/9/docs/api/java/lang/Boolean.html#parseBoolean-java.lang.String-) as truthy (case-insensitive).

Reply:
Please fill another bug or do a or to fix it

Reply:
Done, issue #7717 . Thanks!